### ISSUE TITLE
API: discussion about supported content-type

### ISSUE THREAD

**Issue by HugoGresse** [Instant-Visio/InstantVisio-WebApp#349]
As of today, the only official spported `content-type` for api request is `x-www-form-urlencoded`. 

The fact that some endpoint work when sending json is a side effect. 

Some of our client will probably not have up to date SI and supporting `x-www-form-urlencoded` is the minimum we can do to help them. 

So the question is should we support both json and form input data?

**Comment by MattiaPrimavera** [comment]
- I think I agree more with [this post](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/) stating the best practice should be to accept `json` in both `request/response`:

    > REST APIs should accept JSON for request payload and also send responses to JSON. JSON is the standard for transferring data. Almost every networked technology can use it: JavaScript has built-in methods to encode and decode JSON either through the Fetch API or another HTTP client. Server-side technologies have libraries that can decode JSON without doing much work.

- In case a client has some issues with that, we can choose later how to solve the issue (for my knowledge anyway, which SI cannot send `json` content-types btw?)
- also this way we would have a consistent `content-type` between **request** and **response**, since we send back `json` to the client side

**Comment by HugoGresse** [comment]
I believe some old java 6 code base may have difficulties, on top of that if they are not used to it may cause friction. Supporting a list form really seems to be required, as I stated, the question is  should we support both json and form input data?

**Comment by ntchou** [comment]
I agree with @MattiaPrimavera. Support only `json` and if we have some issues we'll see at this moment

**Comment by HugoGresse** [comment]
I persist with the `form` content type. I will keep this as it works as of today and will need to support both in the future.

**Comment by MattiaPrimavera** [comment]
If you really want to stick with `form` type, we have to change the way of sending POST requests on the front-end side (doable as well)

**Comment by HugoGresse** [comment]
you could keep it as it works currently knowing specific case for array.

**Comment by MattiaPrimavera** [comment]
Shouldn't you expect url encoded values if you accept `x-www-form-urlencoded` by the way ? You're currently expecting to receive a `stringified` version of participant addresses, not a `urlencoded` one. So I fixed the call by `stringifying` the destinations to send to the API (on client side), and it works by still using a `json` content-type: #361
